pkgcheck

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+branch=$1
+arch=$(dpkg --print-architecture)
+
+case "${branch}" in
+	"xenial"*) repo="repo.ubports.com";;
+	*) repo="repo2.ubports.com";;
+esac
+
+readarray -t pkgs < <(awk '/Package/{print $2}' /var/lib/apt/lists/${repo}_dists_${branch//_/%5f}_main_binary-${arch}_Packages|uniq)
```

Comment:
Is there a reason why you are storing this into a variable instead of passing it directly to the loop below?

Something like:

```sh
while read pkg
do
    ...
done < <(awk '/Package/{print $2}' \
    /var/lib/apt/lists/${repo}_dists_${branch//_/%5f}_main_binary-${arch}_Packages|uniq)
```

You also don't need the `relpkgs` variable, but you can call `echo -n "${pkg} "` inside the loop for each installed package.

Reply:
i wasn't aware this is possible, i would need to test how differently it behaves.